client/allocrunner/taskrunner/template_hook.go

```diff
@@ -113,6 +116,10 @@ func (h *templateHook) Prestart(ctx context.Context, req *interfaces.TaskPrestar
 	h.vaultToken = req.VaultToken
 	h.nomadToken = req.NomadToken
 
+	if req.Task.Vault != nil {
+		h.vaultConfig = h.config.clientConfig.VaultConfigs[req.Task.Vault.Cluster]
+	}
+
 	// Set vault namespace if specified
 	if req.Task.Vault != nil {
```

Comment:
Can we move this under the same conditional block as above?

Reply:
I will moved the template changes to a new PR, but I will make sure this gets addressed.

Reply:
Applied in https://github.com/hashicorp/nomad/commit/65852f405f46f1bb2d0511252338ba2b9301304e#diff-23a9c60db0b49fdf4f4e68fd78ac41ba1eeaccf2b463b9898d25beb7632dc24fL116-L119